src/videojs.ads.js

```diff
@@ -477,11 +460,33 @@ var
             }
           },
           'content-playback': {
+            enter: function() {
+              player.trigger({
+                type: 'contentplayback',
+                triggerevent: fsm.triggerevent
+              });
+              // Make sure that the cancelPlayTimeout is cleared
+              if (player.ads.cancelPlayTimeout) {
+                clearImmediate(player.ads.cancelPlayTimeout);
+                player.ads.cancelPlayTimeout = null;
+                // Trigger playback if play was canceled and the player is paused.
+                if (player.paused()) {
+                  player.play();
+                }
+              }
+            },
             events: {
+              // in the case of a timeout, adsready might come in late.
+              'adsready': function() {
+                if (player.paused()) {
+                  this.state = 'ads-ready';
+                } else {
+                  this.state = 'preroll?';
```

Comment:
This code was in the ad-timeout-playback state.  I was just porting it over.  I think your suggestion makes it harder for ad-integrations cause then they need to know whether they have missed the loop or not with regards to the timing of them triggering ads-ready.